File: Serene3/Serene3.Web/Modules/com/productimage/product/productColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Serene3.com.Columns;

[ColumnsScript("com.product")]
[BasedOnRow(typeof(productRow), CheckNames = true)]
public class productColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string CeitoName { get; set; }
    public string Photo { get; set; }
    public string Type { get; set; }
    public string Siplername { get; set; }
    public DateOnly Entitydata { get; set; }
    public string Reviewurl { get; set; }
    public string Productquntity { get; set; }
    public double Rating { get; set; }
    public int ReviewCount { get; set; }
}

File: Serene3/Serene3.Web/Modules/ServerTypes/com/productColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { productRow } from "./productRow";

export interface productColumns {
    Id: Column<productRow>;
    Name: Column<productRow>;
    Description: Column<productRow>;
    Price: Column<productRow>;
    CeitoName: Column<productRow>;
    Photo: Column<productRow>;
    Type: Column<productRow>;
    Siplername: Column<productRow>;
    Entitydata: Column<productRow>;
    Reviewurl: Column<productRow>;
    Productquntity: Column<productRow>;
    Rating: Column<productRow>;
    ReviewCount: Column<productRow>;
}

export class productColumns extends ColumnsBase<productRow> {
    static readonly columnsKey = 'com.product';
    static readonly Fields = fieldsProxy<productColumns>();
}